### PR TITLE
feat(addresses): tag address as favorite

### PR DESCRIPTION
**Pull request by vkrashenko**
## Description
**Acceptance:**
Customer/Client App can tag one or more of my addresses to "favorite" list”.
If Customer/Client views a full list of all shipping addresses in checkout, but the tagged addresses appear at the top of the list and IsFavorite  property set to True.
Tag Favorites is unique per a customer.  
I can tag for my and my organization addresses.
If I delete address, the favorites address should be deleted from persistent storage.

### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-434
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.51.0-pr-964-7084-7084cd35.zip

### CODE REVIEW

File: client-app/pages/company/info.vue [hunk 1]
```diff
@@ -172,6 +172,32 @@
 
             <template #desktop-body>
               <tr v-for="address in paginatedAddresses" :key="address.id" class="even:bg-gray-50">
+                <td class="cursor-pointer px-4 py-3 text-center">
+                  <VcTooltip class="ml-1 !block" placement="bottom-start" strategy="fixed">
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:176]: "!block" for what?

File: client-app/pages/company/info.vue [hunk 2]
```diff
@@ -172,6 +172,32 @@
...
+                    <template #trigger>
+                      <VcIcon
+                        :class="{
+                          'text-[--color-neutral-400]': !address.isFavorite,
+                          'text-[--color-primary-500]': address.isFavorite,
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:181]: Merge 'dev' branch => use "text-neutral-400", "text-primary" classes

File: client-app/pages/company/info.vue [hunk 3]
```diff
@@ -172,6 +172,32 @@
...
+                    </template>
+
+                    <template #content>
+                      <div
+                        v-html-safe="
+                          address.isFavorite
+                            ? $t('pages.company.info.remove_from_favorite')
+                            : $t('pages.company.info.add_to_favorite')
+                        "
+                        class="w-44 rounded-sm px-3.5 py-1.5 text-11 font-light shadow-sm-x-y"
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:196]: 1. text-11, shadow-sm-x-y classes are deprecated (See tailwind.config.ts).
2. check mobile view - 320px - 375px (w-44 looks weird).

**Review by goldenmaya — CHANGES_REQUESTED** [review]

File: client-app/pages/company/info.vue [hunk 4]
```diff
@@ -172,6 +172,32 @@
...
+                        }"
+                        name="star"
+                        size="md"
+                        @click="toggleFavoriteAddress(address.isFavorite, address.id)"
+                      />
+                    </template>
+
+                    <template #content>
+                      <div
+                        v-html-safe="
```
Comment (asvishnyakov) [inline client-app/pages/company/info.vue:191]: There is nothing to sanitize, so it's redudant

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: locales/en.json [hunk 5]
```diff
@@ -1072,6 +1072,8 @@
         "no_addresses_message": "There are no addresses yet",
         "address_not_delete_message": "The default address cannot be deleted",
         "address_saved_successful_message": "The address has been successfully saved",
+        "add_to_favorites": "Mark as favorite",
+        "remove_from_favorites": "Remove from favorite ",
```
[the remark this answers is not in the document]
Comment (vkrashenko) [reply locales/en.json:1076]: image

**Comment by goldenmaya** [comment]
Screenshot 2024-02-14 at 12 59 37
It's imposible to check layout.

File: client-app/pages/company/info.vue [hunk 6]
```diff
@@ -131,13 +131,25 @@
                   </span>
                 </div>
 
-                <div v-if="userCanEditOrganization" class="absolute right-4 top-3">
+                <div class="absolute right-4 top-3 flex flex-col items-center">
                   <AddressDropdownMenu
+                    v-if="userCanEditOrganization"
                     :address="item"
                     placement="left-start"
+                    class="flex items-center"
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:139]: Are these changes necessary?

File: client-app/pages/company/info.vue [hunk 7]
```diff
@@ -131,13 +131,25 @@
...
                     @edit="openAddOrUpdateCompanyAddressModal(item)"
                     @delete="openDeleteAddressModal(item)"
                   />
+                  <VcIcon
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:143]: Add empty line between tags

File: locales/en.json (hunk 5 above)
Comment (artem-dudarev) [inline locales/en.json:1076]: Remove trailing space.
Change `favorite` to `favorites`

**Review by asvishnyakov — APPROVED** [review]

File: client-app/pages/company/info.vue (hunk 6 above)
Comment (vkrashenko) [reply client-app/pages/company/info.vue]: fixed

**Review by goldenmaya — APPROVED** [review]

**Pull request merged**